### PR TITLE
enforce proper range for rent burn_percent

### LINKED ISSUE #7189: Standardise rent burn percentage
#### Problem
Currently, rent burn percent is in range of 0 to 255 (`std::u8::Max`), which is unnecessary, and offers little advantage over standard percentage range of 0 to 100.

#### Proposed Solution
 `burn_percent`  need to have standard percentage range.

### PR DESCRIPTION
**Pull request by ParthDesai**
#### Problem
Currently, rent burn percent is defined to be in range of 0 to 255 (std::u8::Max), which is unnecessary, and offers little advantage over standard percentage range of 0 to 100.

#### Summary of Changes
proper range for `burn_percent` is enforced.

Fixes #7189

### CODE REVIEW

File: genesis/src/main.rs [hunk 1]
```diff
@@ -333,11 +333,19 @@ fn main() -> Result<(), Box<dyn error::Error>> {
...
     let faucet_pubkey = pubkey_of(&matches, "faucet_pubkey_file");
 
-    let rent = Rent {
-        lamports_per_byte_year: value_t_or_exit!(matches, "lamports_per_byte_year", u64),
-        exemption_threshold: value_t_or_exit!(matches, "rent_exemption_threshold", f64),
-        burn_percent: value_t_or_exit!(matches, "rent_burn_percentage", u8),
-    };
+    let rent_burn_percentage = value_t_or_exit!(matches, "rent_burn_percentage", u8);
+    if rent_burn_percentage > 100 {
+        panic!(
```
Comment (mvines) [inline genesis/src/main.rs]: Let's never panic on user input, instead provide a clean exit for the user to retry.   In this case, a clap.rs `.validator()` attached to the rent_burn_percentage arg would be ideal, eg: 
https://github.com/solana-labs/solana/blob/d5c8b26a458bc469229640b359aa05339ca66c38/gossip/src/main.rs#L69

File: runtime/src/bank.rs [hunk 2]
```diff
@@ -1706,11 +1707,8 @@ mod tests {
             &dummy_leader_pubkey,
             dummy_leader_lamports,
         );
-        genesis_config.rent = Rent {
-            lamports_per_byte_year: 5,
-            exemption_threshold: 1.2,
-            burn_percent: 5,
-        };
+
+        genesis_config.rent = Rent::new(5, 1.2, 5);
```
Comment (mvines) [inline runtime/src/bank.rs]: I don't think this is a readability improvement.  We go from a list of descriptive name/value pairs to an unreadable string of 3 numbers.

File: sdk/src/rent.rs [hunk 3]
```diff
@@ -10,7 +10,7 @@ pub struct Rent {
     pub exemption_threshold: f64,
 
     // What portion of collected rent are to be destroyed, percentage-wise
-    pub burn_percent: u8,
+    burn_percent: u8,
```
Comment (mvines) [inline sdk/src/rent.rs]: Why de-pub and add `get_burn_percentage()`, making `burn_percent` unlike the other two fields?

File: genesis/src/main.rs (hunk 1 above)
Comment (ParthDesai) [reply genesis/src/main.rs]: cool. Will do that.

File: sdk/src/rent.rs (hunk 3 above)
Comment (ParthDesai) [reply sdk/src/rent.rs]: @mvines It was done to make sure, that it cannot be modified arbitrary.

**Comment by ParthDesai** [comment]
@mvines Overall, I did it to make sure, it shouldn't be ever possible to make a rent object with invalid `burn_percent`, but I guess it is enough if we just add validation at genesis side.

File: sdk/src/rent.rs (hunk 3 above)
Comment (mvines) [reply sdk/src/rent.rs]: The other two field would have that same concern though, and the downside of removing pub from only one is inconsistency and possible confusion.   I'd restore the `pub` here, and if we want to "clean up" the entire struct then do that as a separate PR (IMO it's not worth it though)

**Comment by ParthDesai** [comment]
@mvines How does it look now?

**Comment by ParthDesai** [comment]
> https://github.com/solana-labs/solana/blob/d5c8b26a458bc469229640b359aa05339ca66c38/sdk/src/fee_calculator.rs#L126
> 
> needs an update. Ideally this PR would touch a test or two as well, otherwise it shows we have poor test coverage around fee burning

@mvines This PR focuses on rent only. If you want, I can add another PR for tx fee's burn percent.

**Comment by mvines** [comment]
Ack, yes please.  Multiple `burn_percent` fields with different semantics 🤕

**Pull request merged**